src/server/replica.cc

```diff
@@ -402,59 +402,88 @@ error_code Replica::Greet() {
   // Note that we currently do not support dragonfly->redis replication.
   RETURN_ON_ERR(SendCommand("REPLCONF capa dragonfly", &serializer));
   RETURN_ON_ERR(ReadRespReply(&io_buf, &consumed));
-  if (!CheckRespFirstTypes({RespExpr::STRING})) {
+  auto err_handler = [&io_buf]() {
+    LOG(ERROR) << "Unexpected response " << ToSV(io_buf.InputBuffer());
     return make_error_code(errc::bad_message);
-  }
+  };
+
+  if (!CheckRespFirstTypes({RespExpr::STRING}))
+    return err_handler();
 
   string_view cmd = ToSV(resp_args_[0].GetBuf());
+
   if (resp_args_.size() == 1) {  // Redis
-    if (cmd != "OK") {
-      LOG(ERROR) << "Unexpected response " << cmd;
-      return make_error_code(errc::bad_message);
-    }
-  } else if (resp_args_.size() == 3) {  // it's dragonfly master.
-    // Response is: <master_repl_id, syncid, num_shards>
-    if (!CheckRespFirstTypes({RespExpr::STRING, RespExpr::STRING, RespExpr::INT64}) ||
-        resp_args_[0].GetBuf().size() != CONFIG_RUN_ID_SIZE) {
-      LOG(ERROR) << "Unexpected response " << ToSV(io_buf.InputBuffer());
-      return make_error_code(errc::bad_message);
-    }
+    if (cmd != "OK")
+      return err_handler();
+  } else if (resp_args_.size() >= 3) {  // it's dragonfly master.
+    if (auto ec = HandleCapaDflyResp(); ec)
+      return err_handler();
+    io_buf.ConsumeInput(consumed);
```

Comment:
why do you need this here?

Reply:
Don't need it (previously the buffer was shared with the part in `ConfigureDflyMaster` so it was needed)